### PR TITLE
Adding hasSize file assertion

### LINKED ISSUE #1311: Additional file assertions?
#### Summary

I'd like to contribute some additional file assertions to core (to check the mode/ownership and other file attributes) if that's something that the core team finds useful.

I'm happy to submit a PR if this is something you find useful.

#### Example

```java
assertThat(someFile)
  .hasMode(....)
  .isExecutable()
  .isReadOnly()
```

### PR DESCRIPTION
**Pull request by ChaithanyaGK**
#### Check List:
* Fixes #1311 (planning to raise one PR per assertion)
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

### CODE REVIEW

File: src/main/java/org/assertj/core/api/AbstractFileAssert.java [hunk 1]
```diff
@@ -338,6 +338,31 @@ public SELF hasBinaryContent(byte[] expected) {
...
+   * assertThat(file).hasSize(0);
+   * assertThat(file).hasSize(1);</code></pre>
+   *
+   * @param expected the expected {@code File}'s size in bytes.
+   * @return {@code this} assertion object.
+   * @throws AssertionError if the actual {@code File} is {@code null}.
+   * @throws AssertionError if the actual {@code File} is not an existing file.
+   * @throws AssertionError if the size of the actual {@code File} is not equal to the given size.
+   */
+  public SELF hasSize(long expected) {
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractFileAssert.java]: Rename to `expected` to `expectedInBytes`, I wonder if we should rename the assertion `hasSizeInBytes` thoughts ?

File: src/main/java/org/assertj/core/api/AbstractFileAssert.java [hunk 2]
```diff
@@ -338,6 +338,31 @@ public SELF hasBinaryContent(byte[] expected) {
...
+   * <p>
+   * Example:
+   * <pre><code class='java'> File file = File.createTempFile(&quot;tmp&quot;, &quot;bin&quot;);
+   * Files.write(file.toPath(), new byte[] {1, 1});
+   *
+   * // assertion will pass
+   * assertThat(file).hasSize(2);
+   *
+   * // assertions will fail
+   * assertThat(file).hasSize(0);
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/api/AbstractFileAssert.java]: keep one failing assertion as the second one does not add much value

File: src/main/java/org/assertj/core/error/ShouldHaveSize.java [hunk 3]
```diff
@@ -14,13 +14,17 @@
...
+import java.io.File;
+
 /**
  * Creates an error message indicating that an assertion that verifies that a value have certain size failed.
  * 
  * @author Alex Ruiz
  */
 public class ShouldHaveSize extends BasicErrorMessageFactory {
 
+  private static final String SHOULD_HAVE_FILE_SIZE = "%nExpecting file%n  <%s>%nto have size:%n  <%s>%nbut had:%n  <%s>";
```
Comment (joel-costigliola) [inline src/main/java/org/assertj/core/error/ShouldHaveSize.java]: let's change the error a bit to:
```
 "%nExpecting file%n  <%s>%nto have a size of:%n  %s bytes%nbut had:%n  %s";
```

File: src/test/java/org/assertj/core/error/ShouldHaveSize_create_Test.java [hunk 4]
```diff
@@ -48,4 +52,19 @@ public void should_create_error_message_with_hexadecimal_representation() {
...
+    ErrorMessageFactory factory = shouldHaveSize(new FakeFile("abc"), 3L);
+    String expectedErrorMessage = format("[Test] %n"
+                                         + "Expecting file%n"
+                                         + "  <abc>%n"
+                                         + "to have size:%n"
+                                         + "  <3L>%n"
+                                         + "but had:%n"
+                                         + "  <0L>");
+    // THEN
+    assertThat(factory.create(new TextDescription("Test"), new StandardRepresentation())).isEqualTo(expectedErrorMessage);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/error/ShouldHaveSize_create_Test.java]: extract `factory.create(new TextDescription("Test"), new StandardRepresentation())` to a variable and put it in the `// WHEN` section.

**Comment by joel-costigliola** [comment]
@ChaithanyaGK PR reviewed, good work!

File: src/main/java/org/assertj/core/api/AbstractFileAssert.java (hunk 1 above)
Comment (ChaithanyaGK) [reply src/main/java/org/assertj/core/api/AbstractFileAssert.java]: Makes sense, will refactor.

**Comment by ChaithanyaGK** [comment]
@joel-costigliola Thanks for the review. I've addressed your comments.

**Comment by joel-costigliola** [comment]
After getting users feedback, we decided to go back to `hasSize` (sorry for the unnecessary work), which I have done with other minor improvements but nice work, appreciated @ChaithanyaGK!

**Pull request closed without merging**